pkg/inject/sidecar.go

```diff
@@ -409,3 +411,16 @@ func joinTags(tags map[string]string) string {
 	sort.Strings(tagsSlice)
 	return strings.Join(tagsSlice, ",")
 }
+
+func getContainerName(containers []corev1.Container, agentIdx int) string {
```

Comment:
Now that it's factored out, it should now be easy to add tests for all conditions here ;-)

Reply:
Tests added